src/main/java/com/google/sps/servlets/Login.java

```diff
@@ -0,0 +1,38 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Sends back the login status of User and a login/logout link.
+ */
+@WebServlet("/login-status")
+public class Login extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    response.setContentType("application/json");
+
+    String page = request.getParameter("page");
```

Comment:
Can you explain what page a little? I get that it's the redirection page but where does it come from?

Reply:
Since this function handles GET requests, the page parameter comes from the URL itself. A request to this servlet might look like /login-status?page=projects, which would mean the servlet would return a login/logout link that, once clicked, would send the user to the projects page after logging them in or out.